integration/cluster_test.go

```diff
@@ -416,3 +419,77 @@ func clusterMustProgress(t *testing.T, membs []*member) {
 		mcancel()
 	}
 }
+
+func TestTransferLeader(t *testing.T) {
```

Comment:
For 3), if the leader is killed with `etcdserver.Stop`, there's no need to wait on the `StopNotify()` since it already waits on the done channel.

For 4,5), I think the timing will be problematic on CI.

Is there a reason why sending Puts until the transfer completes wouldn't work (e.g., it's expected that pending puts will be lost on transfer)? I think the continual put case is what would be expected from a clean transfer and probably the most reliable way to test the functionality works as intended.